src/numerical-difference.cpp

```diff
@@ -0,0 +1,180 @@
+
+#define ALL_OUTPUT_SIGNALS  m_x_filteredSOUT << m_dxSOUT  << m_ddxSOUT
+
+      namespace dynamicgraph = ::dynamicgraph;
+      using namespace dynamicgraph;
+      using namespace dynamicgraph::command;
+      using namespace metapod;
+      using namespace Eigen;
+
+      /// Define EntityClassName here rather than in the header file
+      /// so that it can be used by the macros DEFINE_SIGNAL_**_FUNCTION.
+      typedef NumericalDifference EntityClassName;
+
+      /* --- DG FACTORY ------------------------------------------------------- */
+      DYNAMICGRAPH_FACTORY_ENTITY_PLUGIN(NumericalDifference,"NumericalDifference");
+
+      /* --- CONSTRUCTION ----------------------------------------------------- */
+      /* --- CONSTRUCTION ----------------------------------------------------- */
+      /* --- CONSTRUCTION ----------------------------------------------------- */
+      NumericalDifference::
+      NumericalDifference( const std::string & name )
+        : Entity(name),
+          CONSTRUCT_SIGNAL_IN(x,                dynamicgraph::Vector)
+        ,CONSTRUCT_SIGNAL_OUT(x_filtered,       dynamicgraph::Vector, m_x_dx_ddxSINNER)
+        ,CONSTRUCT_SIGNAL_OUT(dx,               dynamicgraph::Vector, m_x_dx_ddxSINNER)
+        ,CONSTRUCT_SIGNAL_OUT(ddx,              dynamicgraph::Vector, m_x_dx_ddxSINNER)
+        ,CONSTRUCT_SIGNAL_INNER(x_dx_ddx,       dynamicgraph::Vector, m_xSIN)
+      {
+        Entity::signalRegistration( ALL_INPUT_SIGNALS << ALL_OUTPUT_SIGNALS);
+        
+        /* Commands. */
+        addCommand("getTimestep",
+                   makeDirectGetter(*this,&m_dt,
+                                    docDirectGetter("Control timestep [s ]","double")));
+        addCommand("getDelay",
+                   makeDirectGetter(*this,&m_delay,
+                                    docDirectGetter("Delay in the estimation of signal x","double")));
+        addCommand("init", makeCommandVoid2(*this, &NumericalDifference::init,
+                              docCommandVoid2("Initialize the estimator.",
+                                              "Control timestep [s].",
+                                              "Estimation delay for signal x")));
+      }
+
+
+      /* --- COMMANDS ---------------------------------------------------------- */
+      /* --- COMMANDS ---------------------------------------------------------- */
+      /* --- COMMANDS ---------------------------------------------------------- */
+      void NumericalDifference::init(const double &timestep, const double& delay)
+      {
+        assert(timestep>0.0 && "Timestep should be > 0");
+        assert(delay>=1.5*timestep && "Estimation delay should be >= 1.5*timestep");
+        m_dt = timestep;
+        m_delay = delay;
+        int winSizeEnc     = (int)(2*delay/m_dt);
+        assert(winSizeEnc>=3 && "Estimation-window's length should be >= 3");
+
+        m_filter         = new QuadEstimator(winSizeEnc, N_JOINTS, m_dt);
```

Comment:
You shouldn't use N_JOINTS as size, but you should take the signal size as input of the init command